resources/testdriver.js

```diff
@@ -23,45 +23,71 @@
         }
     }
 
+    function getCenterPoint(rect) {
+        const x = 0.5 * (rect.left + rect.right);
+        const y = 0.5 * (rect.top + rect.bottom);
+
+        return [x, y];
+    }
+
     function getInViewCenterPoint(rect) {
-        var left = Math.max(0, rect.left);
-        var right = Math.min(window.innerWidth, rect.right);
-        var top = Math.max(0, rect.top);
-        var bottom = Math.min(window.innerHeight, rect.bottom);
+        const left = Math.max(0, rect.left);
+        const right = Math.min(window.innerWidth, rect.right);
+        const top = Math.max(0, rect.top);
+        const bottom = Math.min(window.innerHeight, rect.bottom);
 
-        var x = 0.5 * (left + right);
-        var y = 0.5 * (top + bottom);
+        const x = 0.5 * (left + right);
+        const y = 0.5 * (top + bottom);
 
         return [x, y];
     }
 
-    function getPointerInteractablePaintTree(element) {
-        let elementDocument = element.ownerDocument;
+    function rectConstrainedToViewport(rect) {
+        const left = Math.max(0, rect.left);
+        const right = Math.min(window.innerWidth, rect.right);
+        const top = Math.max(0, rect.top);
+        const bottom = Math.min(window.innerHeight, rect.bottom);
+
+        return new DOMRect(rect.x, rect.y, Math.max(right - left, 0), Math.max(bottom - top, 0));
+    }
+
+    function getElementListAtVisibleCenter(element) {
```

Comment:
This function was intended to match https://w3c.github.io/webdriver/#dfn-pointer-interactable-paint-tree, and named as such.

I'm not saying we can't deviate from WebDriver here, but it does hint at there maybe being issues in the WebDriver definitions here.